dynamiqs/time_tensor.py

```diff
@@ -239,3 +278,170 @@ def __add__(self, other: Tensor | TimeTensor) -> TimeTensor:
+
+        if t < self.times[0] or t >= self.times[-1]:
+            return static  # (..., n, n)
+        else:
+            # find the index $k$ such that $t \in [t_k, t_{k+1})$
+            idx = torch.searchsorted(self.times, t, side='right') - 1
+            v = self.values[:, idx, ...]  # (npwc, ...)
+            v = v.permute(*range(1, v.ndim), 0)  # (..., npwc)
+            v = v.view(*v.shape, 1, 1)  # (..., npwc, n, n)
+            return (v * self.tensors).sum(-3) + static  # (..., n, n)
+
+    def view(self, *shape: int) -> TimeTensor:
+        # shape: (..., n, n)
+        values = self.values.view(self.npwc, self.nv, *shape[:-2])
+        return PWCTimeTensor(self.times, values, self.tensors, static=self.static)
+
+    def adjoint(self) -> TimeTensor:
+        return PWCTimeTensor(
+            self.times, self.values.conj(), self.tensors.mH, static=self.static.mH
+        )
+
+    def __neg__(self) -> TimeTensor:
+        return PWCTimeTensor(
+            self.times, self.values, -self.tensors, static=-self.static
+        )
+
+    def __mul__(self, other: Number | Tensor) -> TimeTensor:
+        return PWCTimeTensor(
+            self.times, self.values, self.tensors * other, static=self.static * other
+        )
+
+    def __add__(self, other: Tensor | TimeTensor) -> TimeTensor:
+        if isinstance(other, Tensor):
+            static = self.static + other
+            return PWCTimeTensor(self.times, self.values, self.tensors, static=static)
+        elif isinstance(other, ConstantTimeTensor):
+            return self + other.tensor
+        elif isinstance(other, PWCTimeTensor):
+            # merge times and values -> times: (nv+1), values: (npwc1 + npwc2, nv, ...)
+            t1, t2 = self.times, other.times
+            v1, v2 = self.values, other.values
+            times, values = _merge_pwc_times_values(t1, t2, v1, v2)
+
+            # merge tensors -> tensors: (npwc1 + npwc2, n, n)
+            tensors = torch.cat((self.tensors, other.tensors))
+
+            # merge static part -> static: (n, n)
+            static = self.static + other.static
+
+            return PWCTimeTensor(times, values, tensors, static=static)
+        else:
+            return NotImplemented
+
+
+def _merge_pwc_times_values(t1, t2, v1, v2):
+    # t1: (nv1+1)
+    # t2: (nv2+1)
+    # v1: (npwc1, nv1, ...)
+    # v2: (npwc1, nv2, ...)
+    # --> (nv+1), (npwc1 + npwc2, nv, ...)
+
+    # Example with (...) = () and npwc1 = npwc2 = 1:
+    # t1 = [0, 10, 20]
+    # v1 = [[1, 2]]
+    # t2 = [0, 5, 10, 30]
+    # v2 = [[1, 2, 3]]
+    # times: 0     5     10        20        30 = [0, 5, 10, 20, 30]
+    #        |-----|-----|---------|---------|
+    # v1n:   |  1  |  1  |    2    |    0    |  = [1, 1, 2, 0]
+    # v2n:   |  1  |  2  |    3    |    3    |  = [1, 2, 3, 3]
+    # t = [0, 5, 10, 20, 30]
+    # v = [[1, 1, 2, 0], [1, 2, 3, 3]]
+
+    # merge times
+    t = torch.cat((t1, t2)).sort()[0].unique(sorted=True)  # (nv+1)
```

Comment:
```suggestion
    t = torch.cat((t1, t2)).unique(sorted=True)  # (nv+1)
```
Isn't this already sorting?

Reply:
Yep 👍